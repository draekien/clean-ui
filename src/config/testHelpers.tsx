import React, { ReactNode } from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from '../components/theme/theme.provider';

export const themedRender = (element: ReactNode) =>
  render(<ThemeProvider>{element}</ThemeProvider>);
