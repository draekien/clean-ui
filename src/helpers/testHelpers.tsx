import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from '../components/theme/theme.provider';

export const themedRender = (element: React.ReactNode) => {
  return render(<ThemeProvider>{element}</ThemeProvider>);
};

export const hexToRgb = (hex: string) => {
  hex = hex.replace(/[^0-9A-F]/gi, '');
  const bigint = parseInt(hex, 16);
  const r = (bigint >> 16) & 255;
  const g = (bigint >> 8) & 255;
  const b = bigint & 255;

  return `rgb(${r}, ${g}, ${b})`;
};
