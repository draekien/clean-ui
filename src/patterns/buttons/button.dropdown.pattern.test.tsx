import React from 'react';
import { themedRender } from '../../helpers/testHelpers';
import { DropdownButton } from '../..';

describe('Dropdown button component', () => {
  it('should match snapshot', () => {
    const { container } = themedRender(
      <DropdownButton
        value="select"
        items={{
          rows: [
            'Travis Willingham',
            'Laura Bailey',
            'Ashleigh Johnson',
            'Sam Riegal',
            'Matthew Mercer',
            'Marisha Ray',
            "Liam O'Brian",
            'Talisen Jaffe',
            'Percival Fredrickstein von Musel Klossowski de Rolo III',
          ],
        }}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
