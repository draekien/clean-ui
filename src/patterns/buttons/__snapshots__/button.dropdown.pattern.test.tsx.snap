// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown button component should match snapshot 1`] = `
<div>
  <div>
    <div
      class="css-1rxpneb-DropdownButton"
    >
      <button
        class="css-115rw74-Button"
        icon="expand_more"
      >
        <div
          class="css-1xz4u2o-Text css-vurnku"
        >
          select
        </div>
        <span
          class="css-lh6qdg-ButtonIcon"
        >
          <i
            class="material-icons  css-y0j37o-Icon"
          >
            expand_more
          </i>
        </span>
      </button>
    </div>
  </div>
</div>
`;
